hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hudi/command/HoodieSparkValidateDuplicateKeyRecordMerger.scala

```diff
@@ -43,5 +45,9 @@ class HoodieSparkValidateDuplicateKeyRecordMerger extends HoodieSparkRecordMerge
   /**
    * The kind of merging strategy this recordMerger belongs to. An UUID represents merging strategy.
    */
-  override def getMergingStrategy: String = HoodieRecordMerger.DEFAULT_MERGER_STRATEGY_UUID
+  override def getMergingStrategy: String = STRATEGY_STR
+}
+
+object HoodieSparkValidateDuplicateKeyRecordMerger {
+  final val STRATEGY_STR = "fb092649-0fdc-4c14-9113-acde3034a6c4"
```

Comment:
Could we make the ID easier to read, e.g., 
"00000000-0000-0000-0000-000000000000": Avro payload based merging;
"00000000-0000-0000-0000-000000000001": overwrite with latest;
"00000000-0000-0000-0000-000000000002": event time ordering / default hoodie payload.  Still reserve the existing ID as an alternative ID;
"00000000-0000-0000-0000-000000000003": validate duplicate key record merger;
etc.

Reply:
Also is it necessary to use UUID format, maybe "0", "1", "2" are shorter IDs.

Reply:
If we want to do this it should be in a followup pr. I think the reasoning is that we don't want collisions for any user created payloads

Reply:
Let's have a Jira ticket to track that.  In the current scheme, it's still hard to avoid collision, since there is no rule on how to generate these IDs.  A general String with namespace and number might be better.

Reply:
HUDI-8422 as a follow-up.